hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/ReadableFreezeUpgradeActions.java

```diff
@@ -220,13 +219,12 @@ private void extractAndReplaceArtifacts(
             FileUtils.cleanDirectory(artifactsDir);
             UnzipUtility.unzip(archiveData.toByteArray(), artifactsLoc);
             log.info("Finished unzipping {} bytes for {} update into {}", size, desc, artifactsLoc);
-            if (desc.equals(PREPARE_UPGRADE_DESC)) {
-                requireNonNull(nodes, "Cannot generate config.txt without a valid list of active nodes");
+            if (nodes != null) {
```

Comment:
Why is this condition changed?

Reply:
Because we have this check earlier:
 final var activeNodes = desc.equals(PREPARE_UPGRADE_DESC) ? allActiveNodes() : null;